src/index.d.ts

```diff
@@ -1,21 +1,41 @@
 import { Properties as CSSProperties } from "csstype";
-import { JSX } from "solid-js";
+import { JSX, Component } from "solid-js";
+
+// This type is only available in Solid 1.5 or later.
+type ValidComponent =
+  | keyof JSX.IntrinsicElements
+  | Component<any>
+  | (string & {});
+
 export interface DefaultTheme {}
 export interface CSSAttribute extends CSSProperties {
   [key: string]: CSSAttribute | string | number | undefined;
 }
+type StylesGenerator<P = {}> =
+  (props: P) => CSSAttribute | string;
+type TagStyleGenerator<P = {}> =
+  (props: P) => number | string | undefined;
+type TagArgs<P = {}> = Array<string | TagStyleGenerator<P>>;
+type StylesArg<P = {}> =
+  | string
+  | CSSAttribute
+  | StylesGenerator<P>
+  | Array<CSSAttribute | StylesGenerator<P>>;
+export declare function keyframes(styles: StylesArg): string;
 export declare function keyframes(
-  tag: TemplateStringsArray | string,
-  ...props: Array<string | number>
+  tag: TemplateStringsArray,
+  ...tagArgs: TagArgs
 ): string;
 export declare function extractCss(): string;
+export declare function glob(styles: StylesArg): void;
 export declare function glob(
-  tag: CSSAttribute | TemplateStringsArray | string,
-  ...props: Array<string | number>
+  tag: TemplateStringsArray,
+  ...tagArgs: TagArgs
 ): void;
+export declare function css(styles: StylesArg): string;
 export declare function css(
-  tag: CSSAttribute | TemplateStringsArray | string,
-  ...props: Array<string | number>
+  tag: TemplateStringsArray,
+  ...tagArgs: TagArgs
 ): string;
```

Comment:
By [having multiple declarations](https://www.typescriptlang.org/docs/handbook/declaration-merging.html#merging-interfaces) we allow multiple arguments only when we use the "tag function", but when we do a regular function call can restirct the type to one argument.

```tsx
// allowed
css`
  color: ${theme.typography.color};
`
// allowed
css({
  color: ${theme.typography.color},
});

// now the following would be disallowed
css(
  {
    color: ${theme.typography.color},
  },
  {
    font-family: ${theme.typography.fontFamily, // this won't be applied
  },
);
```